File: Java/src/com/leetcode/Easy/ReverseLinkedList.java
package com.leetcode.Easy;

import java.util.Stack;

// 206. Reverse Linked List
public class ReverseLinkedList {
    public static class ListNode {
        int val;
        ListNode next;

        ListNode(int x) {
            val = x;
        }
    }

    public ListNode reverseList(ListNode head) {
        if (head == null)
            return null;

        Stack<ListNode> stack = new Stack<>();
        ListNode curr = head;
        while (curr.next != null) {
            stack.push(curr);
            curr = curr.next;
        }

        head = curr;
        while (!stack.isEmpty()) {
            curr.next = stack.pop();
            curr = curr.next;
        }
        curr.next = null;
        return head;
    }
}

File: JavaScript/Easy/ReverseLinkedList.js
/**
 * 206. Reverse Linked List
 * @param {ListNode} head
 * @return {ListNode}
 */
const reverseList = function (head) {
  let curr = head, prev = null;
  while (curr != null) {
    let next = curr.next;
    curr.next = prev;
    prev = curr;
    curr = next;
  }

  return prev;
};